### PR TITLE
Add banner component

### LINKED ISSUE #95: Custom banner in front matter
@samthor commented on [Wed Nov 11 2020](https://github.com/GoogleChrome/dcc-private/issues/247)

We should have support for custom banners on pages as part of our front matter.

Extensions are interested in this for deprecation/update notices as 1st class citizens.

There's a concern that we might sometimes want several banners.

### PR DESCRIPTION
**Pull request by devnook**
Fixes #95

### CODE REVIEW

**Review by MichaelSolati — APPROVED** [review]
LGTM

File: site/_includes/partials/banner.njk [hunk 1]
```diff
@@ -0,0 +1,7 @@
+
+<div class="wrapper banner gap-top-300 rounded-500 banner--{{ banner.type }}">
```
Comment (MichaelSolati) [inline site/_includes/partials/banner.njk:2]: probably should add an if there is a banner for the page before displaying it.

Comment (devnook) [reply site/_includes/partials/banner.njk]: done

File: site/en/style-guide/index.njk [hunk 2]
```diff
@@ -1,5 +1,10 @@
 ---
 title: Style guide
+banner:
+  text: 'This is a banner'
+  action: '/style-guide'
```
Comment (samthor) [inline site/en/style-guide/index.njk:5]: These feel around the wrong way. `action_href` reads to me like it should be the target URL, not the text.

File: site/_includes/partials/banner.njk [hunk 3]
```diff
@@ -0,0 +1,9 @@
+
+{% if banner %}
+  <div class="wrapper banner gap-top-300 rounded-500 banner--{{ banner.type }}">
+    <div class="banner__text">
+      {{ banner.text }}
+    </div>
+    <a class="link" href="{{ banner.action }}">{{ banner.action_href }}</a>
```
Comment (samthor) [inline site/_includes/partials/banner.njk:7]: If these aren't specified, can we hide the link?

**Comment by devnook** [comment]
PTAL

**Review by samthor — APPROVED** [review]

**Comment by robdodson** [comment]
Sorry to jump in late but I think this differs from Jiwoong's designs in the Figma file. @devnook do you have access to that file?

image

**Comment by robdodson** [comment]
Also wanted to note that we have a web component that folks might want to consider using for this called announcement-banner (it's in the _js/third-party directory because it's forked from a different open source project). That banner can be used to set cookies and will hide itself if the user has dismissed it already.

**Pull request merged**